classes/scheduledTask/PKPScheduler.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+/**
+ * @file classes/scheduledTask/PKPScheduler.php
+ *
+ * Copyright (c) 2014-2024 Simon Fraser University
+ * Copyright (c) 2003-2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PKPScheduler
+ *
+ * @brief Scheduler class which is responsible to register scheduled tasks
+ */
+
+namespace PKP\scheduledTask;
+
+use Illuminate\Console\Scheduling\Event;
+use Illuminate\Console\Scheduling\Schedule;
+use PKP\core\PKPContainer;
+use PKP\task\DepositDois;
+use PKP\task\UpdateIPGeoDB;
+use PKP\task\ReviewReminder;
+use PKP\task\ProcessQueueJobs;
+use PKP\task\RemoveFailedJobs;
+use PKP\task\StatisticsReport;
+use PKP\plugins\PluginRegistry;
+use PKP\task\EditorialReminders;
+use PKP\scheduledTask\ScheduledTask;
+use PKP\task\RemoveExpiredInvitations;
+use PKP\scheduledTask\ScheduleTaskRunner;
+use PKP\task\RemoveUnvalidatedExpiredUsers;
+use PKP\plugins\interfaces\HasTaskScheduler;
+
+abstract class PKPScheduler
+{
+    /**
+     * Constructor
+     * 
+     * @param Schedule $schedule The core illuminate Schedule that is responsible to run schedule tasks
+     */
+    public function __construct(protected Schedule $schedule)
+    {
+        $this->schedule = $schedule;
```

Comment:
When you use the "argument promotion" on the constructor, it's not needed to assign the value, it will be done automatically.

```suggestion
```